lib/rubocop/cop/rspec/factory_girl/dynamic_attribute_defined_statically.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryGirl
+        # Prefer declaring dynamic attribute values in a block.
+        #
+        # @example
+        #   # bad
+        #   kind [:active, :rejected].sample
+        #
+        #   # good
+        #   kind { [:active, :rejected].sample }
+        #
+        #   # bad
+        #   closed_at 1.day.from_now
+        #
+        #   # good
+        #   closed_at { 1.day.from_now }
+        #
+        #   # good
+        #   kind :static
+        #
+        #   # good
+        #   comments_count 0
+        #
+        #   # good
+        #   type User::MAGIC
+        class DynamicAttributeDefinedStatically < Cop
+          MSG = 'Use a block to set a dynamic value to an attribute.'.freeze
+
+          def_node_matcher :dynamic_defined_statically?, <<-PATTERN
+          (send nil _ (send ... ))
+          PATTERN
+
+          def_node_search :factory_attributes, <<-PATTERN
+          (block (send nil {:factory, :trait} ...) _ { (begin $...) $(send ...) } )
+          PATTERN
+
+          def on_block(node)
+            return if node.method_name == :trait
+            factory_attributes(node).to_a.flatten.each do |attribute|
+              if dynamic_defined_statically?(attribute)
+                add_offense(attribute, :expression)
+              end
+            end
+          end
+
+          def autocorrect(node)
+            if method_uses_parens?(node.location)
+              autocorrect_replacing_parens(node)
+            else
+              autocorrect_without_parens(node)
+            end
+          end
+
+          private
+
+          def into_factories?(node)
+            node.ancestors.find(&method(:into_factory_block?))
+          end
+
+          def already_defined_in_a_block?(node)
+            node.sibling_index.zero? && node.parent && node.parent.block_type?
+          end
+
+          def method_uses_parens?(location)
+            return false unless location.begin && location.end
```

Comment:
I observed that when the node location has `begin` and `end` filled when the method call has parenthesis. I just observed that this is true and for some reason when we don't put parenthesis the `begin/end` location is not filled. Then I used it as condition. Is this verification weird?? Should I documented it in somehow? 🤔